### ISSUE TITLE
Linq translates into sql error

### ISSUE THREAD

**Issue by quan01994** [linq2db/linq2db#1964]
Linq translates into sql error
code
```c#
var attachmentQry = from att in (from temp in Db.Attachments
                    where temp.Status == 0 && temp.Type == 2
                    select new
                    {
                        temp.ItemId,
                        temp.Url,
                        Index = Sql.Ext.RowNumber().Over().PartitionBy(temp.ItemId).OrderBy(temp.CreateTime).ToValue()
                    })
                where att.Index == 1
                select att;

            var qry = from branchInfo in Db.NotDeletedBranchInfo
                from att in attachmentQry.LeftJoin(m => m.ItemId == branchInfo.BranchId)
                where branchInfo.BranchIsShow == 0
                select new BranchSelectOutput
                {
                    BranchAddress = branchInfo.BranchAddress,
                    BranchId = branchInfo.BranchId,
                    BranchName = branchInfo.BranchName,
                    BusinessHours = branchInfo.BusinessHours,
                    PointX = branchInfo.PointX,
                    PointY = branchInfo.PointY,
                    BranchPic = att.Url
                };
            var list = await qry.ToListAsync();
```
Console print sql
```sql
SELECT
	[m_1].[BranchAddress],
	[m_1].[BranchId],
	[m_1].[BranchName],
	[m_1].[BusinessHours],
	[m_1].[PointX],
	[m_1].[PointY],
	[t1].[Url]
FROM
	[dbo].[BranchInfo] [m_1]
		LEFT JOIN (
			SELECT
				[att].[Url]
			FROM
				(
					SELECT
						ROW_NUMBER() OVER(PARTITION BY [temp_1].[ItemId] ORDER BY [temp_1].[CreateTime]) as [Index_1],
						[temp_1].[ItemId],
						[temp_1].[Url]
					FROM
						[dbo].[Attachment] [temp_1]
					WHERE
						[temp_1].[Status] = 0 AND [temp_1].[Type] = 2
				) [att]
			WHERE
				[att].[Index_1] = 1
		) [t1] ON [att].[ItemId] = [m_1].[BranchId]
WHERE
	[m_1].[BranchIsShow] = 0 AND [m_1].[Status] = 0
```
If I understand it, the translated sql should be like this。
```sql
SELECT
	[m_1].[BranchAddress],
	[m_1].[BranchId],
	[m_1].[BranchName],
	[m_1].[BusinessHours],
	[m_1].[PointX],
	[m_1].[PointY],
	[t1].[Url]
FROM
	[dbo].[BranchInfo] [m_1]
		LEFT JOIN (
			SELECT
				[att].[Url],
				[att].[ItemId]
			FROM
				(
					SELECT
						ROW_NUMBER() OVER(PARTITION BY [temp_1].[ItemId] ORDER BY [temp_1].[CreateTime]) as [Index_1],
						[temp_1].[ItemId],
						[temp_1].[Url]
					FROM
						[dbo].[Attachment] [temp_1]
					WHERE
						[temp_1].[Status] = 0 AND [temp_1].[Type] = 2
				) [att]
			WHERE
				[att].[Index_1] = 1
		) [t1] ON t1.[ItemId] = [m_1].[BranchId]
WHERE
	[m_1].[BranchIsShow] = 0 AND [m_1].[Status] = 0
```

It looks like optimization is over

### Environment details
linq2db version: 2.9.2
Database Server: SqlServer 2008 r2
Database Provider: sqlclient 4.6.0
Operating system: windows
.NET Framework: .net core 2..2

**Comment by quan01994** [comment]
The 2.9.3 version also has the same error

**Comment by sdanyliv** [comment]
Strange, i have fixed similar problem in 2.9.1. Please post classes used in this query. Thanks!

**Comment by quan01994** [comment]
```c#
    public class BranchSelectOutput
    {
        public int BranchId { get; set; }

        public string BranchName { get; set; }

        public string BranchAddress { get; set; }

        public string BusinessHours { get; set; }

        public double Distance { get; set; }

        public string BranchPic { get; set; }

        [JsonIgnore]
        public decimal PointX { get; set; }

        [JsonIgnore]
        public decimal PointY { get; set; }
    }

        [Table(Schema="dbo", Name="Attachment")]
	public partial class AttachmentEntity
	{
		[PrimaryKey, Identity] public int      Id          { get; set; } // int
		
		[Column,     NotNull ] public int      Type        { get; set; } // int
	
		[Column,     NotNull ] public int      ItemId      { get; set; } // int

		[Column,     NotNull ] public string   Name        { get; set; } // nvarchar(128)
	
		[Column,     NotNull ] public string   ContentType { get; set; } // nvarchar(64)

		[Column,     NotNull ] public string   Url         { get; set; } // nvarchar(128)
	
		[Column,     NotNull ] public int      Status      { get; set; } // int
	
		[Column,     NotNull ] public DateTime CreateTime  { get; set; } // datetime
	}

	[Table(Schema="dbo", Name="BranchInfo")]
	public partial class BranchInfoEntity
	{
		[PrimaryKey, Identity] public int      BranchId         { get; set; } // int

		[Column,     NotNull ] public string   BranchCode       { get; set; } // varchar(50)

		[Column,     NotNull ] public string   BranchName       { get; set; } // varchar(50)

		[Column,     NotNull ] public string   BranchParentCode { get; set; } // varchar(50)

		[Column,     NotNull ] public string   CompanyCode      { get; set; } // varchar(32)

		[Column,     NotNull ] public string   AreaCode         { get; set; } // varchar(50)

		[Column,     NotNull ] public string   BranchAddress    { get; set; } // varchar(100)
	
		[Column,     NotNull ] public string   BrandContacts    { get; set; } // varchar(20)

		[Column,     NotNull ] public string   BrandPhone       { get; set; } // varchar(20)

		[Column,     NotNull ] public int      Status           { get; set; } // int

		[Column,     NotNull ] public DateTime CreateTime       { get; set; } // datetime
	
		[Column,     NotNull ] public string   CreateUser       { get; set; } // varchar(50)

		[Column,     NotNull ] public DateTime LastUpdateTime   { get; set; } // datetime

		[Column,     NotNull ] public string   LastUpdateUser   { get; set; } // varchar(50)

		[Column,     NotNull ] public string   BusinessHours    { get; set; } // varchar(100)

		[Column,     NotNull ] public decimal  PointX           { get; set; } // decimal(18, 4)

		[Column,     NotNull ] public decimal  PointY           { get; set; } // decimal(18, 4)

		[Column,     NotNull ] public int      BranchIsShow     { get; set; } // int
		
		[Column,     NotNull ] public int      BranchType       { get; set; } // int
      }

        public IQueryable<BranchInfoEntity> NotDeletedBranchInfo
        {
            get { return this.GetTable<BranchInfoEntity>().Where(m => m.Status == 0); }
        }

```
